### PR TITLE
Fixes: promise usage and string operator usage for recent Sequelize

### PR DESCRIPTION
**Pull request by deksden**
Now module is compatible with Sequelize 4.13+ (did not use string operator aliases)

### CODE REVIEW

**Comment by deksden** [comment]
Tests are passing for me

**Comment by deksden** [comment]
Forgot to mention minor patch:
- mocha now did not exit automatically starting 4.0.0, so added --exit to force exit;

File: lib/backend.js [hunk 1]
```diff
@@ -83,7 +82,7 @@ SequelizeBackend.prototype = {
 			return this.findRow(bucket);
 		}
 		return this.getModel(bucket).findAll({
-			where: {key: {$in: keys.map(function(key){ return key.toString();})}},
+			where: {key: {[this.db.Op.in]: keys.map(function(key){ return key.toString();})}},
```
Comment (yonjah) [inline lib/backend.js]: I think we can save a reference to Op in the constructor and then access it buy `this.Op` might be easier to refactor in the future.

File: lib/backend.js [hunk 2]
```diff
@@ -118,7 +117,7 @@ SequelizeBackend.prototype = {
 		contract(arguments).params('object', 'function').end();
 		// Execute transaction
 		var i =1 ;
-		return Promise.reduce(transaction, function (res, func) {
+		return this.db.Promise.reduce(transaction, function (res, func) {
```
Comment (yonjah) [inline lib/backend.js]: Same as Op save reference in `this.Promise`

File: package.json [hunk 3]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "acl-sequelize",
-  "version": "0.1.2",
+  "version": "0.1.3",
```
Comment (yonjah) [inline package.json]: This has to be a major change. But any way leave the version as is for now

File: package.json [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
   "description": "Node Acl Sequelize Backend",
   "main": "index.js",
   "directories": {
     "test": "test"
   },
   "scripts": {
-    "test": "mocha test/runner.js --reporter spec"
+    "test": "mocha test/runner.js --reporter spec --exit"
```
Comment (yonjah) [inline package.json]: This means we tests aren't cleaning up properly.
Probably need to call self.backend.close() after the test ends (Though v4 had issue where connection wasn't closed properly but I think it was already fixed).
Any way if you have the time you can try see if you can clean it properly but let mocha hang for now

File: package.json [hunk 5]
```diff
@@ -26,19 +26,18 @@
...
-    "acl": "^0.4.7",
-    "chai": "~1.10.0",
-    "mocha": "^2.0.1",
-    "mysql": "^2.6.1",
-    "sequelize": "^3.2.0"
+    "acl": "^0.4.11",
+    "chai": "^4.1.2",
+    "mocha": "^4.0.0",
+    "mysql2": "^1.4.2",
+    "sequelize": "^4.13.2"
```
Comment (yonjah) [inline package.json]: Make sure test version is the same as the peerDependencies one

File: package.json (hunk 3 above)
Comment (deksden) [reply package.json]: You mean leave 0.1.2?

**Comment by deksden** [comment]
All mods are done for now.

**Pull request merged**